processor/elasticapmprocessor/processor.go

```diff
@@ -174,6 +174,15 @@ func (p *MetricProcessor) ConsumeMetrics(ctx context.Context, md pmetric.Metrics
 				ecs.SetHostIP(ctx, resource.Attributes())
 			}
 			p.enricher.Config.Resource.DeploymentEnvironment.Enabled = false
+
+			// Check if resource has a service name for routing decisions
+			hasServiceName := false
+			if serviceName, ok := resource.Attributes().Get(routing.ServiceNameAttributeKey); ok && serviceName.Str() != "" {
+				hasServiceName = true
+			}
+
+			// Route internal metrics to appropriate data streams if needed.
+			routeMetricsToDataStream(resourceMetric.ScopeMetrics(), hasServiceName)
```

Comment:
suggestion: I think we can add a test case to `TestProcessorECS` to validate data_stream for an internal metric.

Reply:
makes sense, thanks!

Reply:
added in https://github.com/elastic/opentelemetry-collector-components/pull/944/changes/1c7b2ad7569eeab33ebbf78ffd2b1056574c01b4